agent/lib/kontena/workers/image_pull_worker.rb

```diff
@@ -4,6 +4,8 @@ class ImagePullWorker
     include Celluloid
     include Kontena::Logging
 
+    exclusive :ensure_image
```

Comment:
The exclusive/caching logic is split across a number of places and hard to follow, wouldn't something like this be equivalent and easier to follow?

```ruby
    def ensure_image(image, deploy_rev, creds = nil)
      exclusive {
        unless fresh_pull?("#{image}:#{deploy_rev}")
          pull_image(image, deploy_rev, creds)

          update_image_cache("#{image}:#{deploy_rev}")
        end
      }
    end
```